tensorflow_serving/batching/batching_session.cc

```diff
@@ -350,8 +369,15 @@ Status BatchingSession::MergeInputTensors(
       const Tensor& tensor = entry.second;
 
       std::vector<Tensor>& tensor_vec = tensors_to_merge[tensor_name];
-      tensor_vec.push_back(tensor);
-
+      Tensor new_tensor = tensor;
+      if (options_.pad_variable_length_inputs) {
+        PaddingResult res = AddPadding(tensor, max_dim_sizes[tensor_name]);
+        if (!res.padding_status.ok()) {
```

Comment:
With the new AddPadding() signature I'm suggesting, you'll be able to use TF_RETURN_IF_ERROR.